command/connect/envoy/envoy.go

```diff
@@ -154,26 +158,74 @@ func (c *cmd) Run(args []string) int {
 		return 1
 	}
 
-	// First argument needs to be the executable name.
+	// Write the Envoy bootstrap config file out to disk in a pocket universe
+	// visible only to the current process (and exec'd future selves).
+	fd, err := writeEphemeralEnvoyTempFile(bootstrapJson)
+	if err != nil {
+		c.UI.Error("Could not write envoy bootstrap config to a temp file: " + err.Error())
+		return 1
+	}
+
+	// On unix systems after exec the file descriptors that we should see:
+	//
+	//   0: stdin
+	//   1: stdout
+	//   2: stderr
+	//   ... any open file descriptors from the parent without CLOEXEC set
+	//
+	// Above we explicitly disabled CLOEXEC for our temp file, so assuming
+	// FD numbers survive across execs, it should just be the value of
+	// `fd`. This is accessible as a file itself (trippy!) under
+	// /dev/fd/$FDNUMBER.
+	magicPath := filepath.Join("/dev/fd", strconv.Itoa(int(fd)))
 
-	// TODO(banks): passing config including an ACL token on command line is jank
-	// - this is world readable. It's easiest thing for now. Temp files are kinda
-	// gross in a different way - we can limit to same-user access which is much
-	// better but we are leaving the ACL secret on disk unencrypted for an
-	// uncontrolled amount of time and in a location the operator doesn't even
-	// know about. Envoy doesn't support reading bootstrap from stdin or ENV
-	envoyArgs := []string{binary, "--config-yaml", bootstrapYaml}
+	// First argument needs to be the executable name.
+	envoyArgs := []string{binary, "--v2-config-only", "--config-path", magicPath}
 	envoyArgs = append(envoyArgs, passThroughArgs...)
 
 	// Exec
-	err = syscall.Exec(binary, envoyArgs, os.Environ())
+	err = unix.Exec(binary, envoyArgs, os.Environ())
 	if err != nil {
 		c.UI.Error("Failed to exec envoy: " + err.Error())
 		return 1
 	}
+
```

Comment:
Can we move all of this code from line 161 to here into a method defined in a separate file that only compiles on `linux` and `darwin`?

The method for other OSes could return a named error like `ErrUnsupportedOS` and then when we get one of those here we can output something like:

> ==> Error: directly running Envoy is only supported on linux and macOS since envoy itself doesn't build on other platforms currently. Use the -bootstrap option to generate the JSON to use when running envoy on a supported OS or via a container or VM.


That way this whole thing should still compile and be somewhat useful on all platforms.